src/components/todoform/todoform.tsx

```diff
@@ -0,0 +1,137 @@
+import { useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import { useTodos } from '../../context/todoProvider';
+import { addTodo, toggleStatus } from '../../api/todos';
+import { USER_ID } from '../../utils/userID';
+import { ErrorType } from '../../types/Error';
+import { Todo } from '../../types/Todo';
+
+export const TodoForm = () => {
+  const {
+    taskName, setTaskName, error, setError, isAddingTask,
+    setIsAddingTask, countIncompleteTask, setTempTodo,
+    todos, setTodos, togglingId, setTogglingId,
+  } = useTodos();
+
+  const isActiveBtnComplededAll = todos.every(task => task.completed);
+
+  const inputRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [todos, error]);
+
+  const handleOnSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setError(null);
+
+    if (taskName.trim().length === 0) {
+      setError(ErrorType.title);
+
+      return;
+    }
+
+    setIsAddingTask(true);
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: taskName.trim(),
+      completed: false,
+    });
+
+    addTodo({
+      userId: USER_ID,
+      title: taskName.trim(),
+      completed: false,
+    })
+      .then((newTask: Todo) => {
+        const newTodo = [...todos, newTask];
+
+        setTodos(newTodo);
+        setTaskName('');
+      })
+      .catch(() => setError(ErrorType.add))
+      .finally(() => {
+        setIsAddingTask(false);
+        setTempTodo(null);
+      });
+  };
+
+  const completeAllTasks = () => {
```

Comment:
Probably could be renamed to something like toggle, cause it's not only completing tasks